File: Java/1.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (map.containsKey(nums[i])) {
                return new int[] { i, map.get(nums[i]) };
            } else {
                map.put(target - nums[i], i);
            }
        }
        return new int[] { -1, -1 };
    }
}

File: Python/1.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = dict()
        for i, num in enumerate(nums):
            remain = target - num
            if remain in map:
                return [i, map[remain]]
            map[num] = i
        return [-1, -1]
